### ISSUE TITLE
Resetting Hoverfly states 

### ISSUE THREAD

**Issue by tommysitu** [SpectoLabs/hoverfly-java#125]
HoverflyRule should always start Hoverfly from a clean state, meaning that there is no existing simulation or journal. 

The following commands should also trigger a full reset: 
```
hoverflyRule.capture('test.json')
hoverflyRule.simulate(classPath('test.json))
```
In addition, `HoverflyRule` should provide manual reset methods:
```
// delete all simulations and journal logs
hoverflyRule.reset(); 

// delete journal logs but reset simulation to the original SimulationSource
hoverflyRule.resetToDefault(); 
```

**Comment by tommysitu** [comment]
There are a few scenarios that could happen:
1. Using `@ClassRule` and simulation is loaded before all the tests, you may need to manually call  `resetJournal` before each test if you need to do verification
2. Using `@ClassRule` and import a new simulation in each test, then reset is triggered by default.
3. Using `@Rule`, you shouldn't need to do anything because Hoverfly is started in clean state for each test

**Comment by tommysitu** [comment]
@mogronalol any thoughts?

**Comment by mogronalol** [comment]
I am actually starting to think we should just remove support for `@Rule`. Just get it to throw an exception and say "Use class rule". It can reset the state behind the scenes for every test, and I can't see any reason to use `@Rule` over `@ClassRule`

Reset can just be called automatically before every test, the user shouldn't have to do anything, whether you import once per test or once per class.

**Comment by tommysitu** [comment]
I don't think you can augment each test method if you are using `@ClassRule`, unless we create a custom `HoverflyTestRunner` that does what you suggested.

**Comment by mogronalol** [comment]
Oh right, so there's no way of using it to do something before each test without a Runner? Bit annoying. I guess that's how we'd get it to work with annotations also. I wonder if this is clashing with the work @lordofthejars is doing with JUnit 5? I've not actually used it that much but am wondering if it changes or replaces the current JUnit Rule mechanism?

**Comment by lordofthejars** [comment]
Hi well in case of JUnit 5, there is a module that you have more or less more control on how everything is going on, so it should not clash. In theory module system is the replacement for runners.

About ClassRules and reset. I think that what you want to do is just reset after each test method with a `ClassRule`. I have never tried at all but probably only using `@Rule` is the way of doing it, I will need to investigate a bit.

In my experience, creating a JUnit runner is really painful for users, because maybe you want to use a parametrized test, or Spring runner or whatever, for this reason I am starting migrating some Arquillian pieces to be `Rule` compatible.

As I said before JUnit 5 should not be a problem.

**Comment by lordofthejars** [comment]
What I don't understand now is the use case of reset, I mean `reset` call should be something to be called not after each method but in a controlled way, just when you want to refresh the simulation file (probably generated by capture mode).

**Comment by mogronalol** [comment]
Hoverfly now stores the history of every single request and response, called a journal. So end the end of the test you will be able to perform verification by making an API call. You don't want this history to be shared between tests, hence the call to `reset()`. See https://github.com/SpectoLabs/hoverfly-java/issues/42

I do think think that we shoudn't need to `reset` the simulation, just the journal maybe?

**Comment by lordofthejars** [comment]
That's my point too, but at the very end just my opinion, maybe offers both
methods.

**Comment by tommysitu** [comment]
Maybe the only explicit reset we support is `resetJournal` and it is used when you need to do verification and you are using @ClassRule and all the tests share the same simulation source.

**Comment by tommysitu** [comment]
Fixed in 0.7.0
